src/test/java/com/example/shopproject/AdminControllerTest.java

```diff
@@ -30,6 +34,39 @@ public class AdminControllerTest {
     @MockBean
     ProductCategoryServiceImpl productCategoryService;
 
+    @Test
+    void postTestProduct() throws Exception{
+        ProductDto product = new ProductDto(4L , "product Test" , "new product" , 2000.0 , 1L);
+
+        this.mockMvc.perform(post("/api/v1/product")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(asJsonString(product)))
+                .andDo(print())
+                .andExpect(status().isOk());
+    }
+
+    @Test
+    void postTestRate() throws Exception{
+        RateDto rate = new RateDto(23L , 1L , 1L ,3.0);
```

Comment:
better to use statis Fixtures class to create objects, and use builder instead of constructor please